fossology/fossology
# fix(copyright): Remove non utf8 strings

## Description

Earlier versions of PostgreSQL used `SQL_ASCII` for default text encoding which stores everything as is and ignore everything that is not ASCII. New versions uses `UTF-8` as default encoding and rejects everything which is not UTF8 compatible.

This PR tries to solve that issue using [icu library](http://site.icu-project.org/home).

Read #1492 for more info.

Since copyright is the only agent which captures text from input file and store it in DB, this PR only removes non UTF-8 strings in copyright agents.

### Changes

1. Use `libicu` to clean matches in copyright agent.

## How to test

1. Install PostgreSQL with UTF-8 as default character encoding for FOSSology.
1. Install the branch and upload a package with many files which allows more chances of copyright being captured with binary strings in between.
1. Check the copyright agent logs from geeky scan view.
1. Check the copyright captured in copyright view.
1. Check if the copyrights appear correctly in reports.

## Notes

After working with ICU library, I realized that we can make the whole copyright agent unicode compatible since the beginning (the regex scan stage). This can be done in a separate PR.

I will open an issue for the same for further discussion before making the changes.

Probable fix for #1492
